File: Program #4 & #5/Program 5/src/Node.java
//Juhwan Lee
//CS202
//Program 5
//This is an implementation of node class for an array of linear linked lists.
//This node class has media data and next pointer.
//This node class has a "Has a" relationship with the media class and uses setters and getters.

public class Node {
    protected Media data;
    protected Node next;
    //default constructor
    public Node(){
        this.data = null;
        this.next = null;
    }
    //setter (data)
    public void set_data(Media data){
        this.data = data;
    }
    //setter (next)
    public void set_next(Node next){
        this.next = next;
    }
    //getter (data)
    public Media get_data(){
        return data;
    }
    //getter (next)
    public Node get_next(){
        return next;
    }
}

File: Program #3/Program #3/node.cpp
//Juhwan Lee
//CS202
//Program 3
//This program is a monster battle program. The user has four monster options and can create
//monsters and confront opponents. Operator overloading and dynamic casting play a very
//important role in this program.

#include "node.h"
using namespace std;

//default constructor for node
node::node()
	: data(NULL), left(NULL), right(NULL)
{
}

//set data
void node::set_data(monster * newdata)
{
	data = newdata;
}

//set left
void node::set_left(node * newleft)
{
	left = newleft;
}

//set right
void node::set_right(node * newright)
{
	right = newright;
}

//get data
monster*& node::get_data()
{
	return data;
}

//get left
node*& node::get_left()
{
	return left;
}

//get right
node*& node::get_right()
{
	return right;
}
